pkg/flow/project.go

```diff
@@ -167,13 +167,32 @@ func (p *Project) ContractConflictExists(network string) bool {
 	return len(all) != len(uniq)
 }
 
+func (p *Project) DefaultHost(network string) string {
+	if network == "" {
+		network = defaultEmulatorNetworkName
+	}
+
+	return p.conf.Networks.GetByName(network).Host
+}
+
+func (p *Project) GetNetworkByName(name string) *config.Network {
+	return p.conf.Networks.GetByName(name)
+}
+
 func (p *Project) Host(network string) string {
 	return p.conf.Networks.GetByName(network).Host
 }
 
-func (p *Project) EmulatorServiceAccount() *config.Account {
+func (p *Project) EmulatorServiceAccount() (*Account, error) {
 	emulator := p.conf.Emulators.GetDefault()
-	return p.conf.Accounts.GetByName(emulator.ServiceAccount)
+	acc := p.conf.Accounts.GetByName(emulator.ServiceAccount)
+	return AccountFromConfig(*acc)
+}
+
+func (p *Project) SetEmulatorServiceKey(privateKey crypto.PrivateKey) {
+	acc := p.accounts[0]
```

Comment:
```suggestion
	acc := p.conf.Accounts.GetByName(emulator.ServiceAccount)
```

I don't think we can assume that the emulator account is always first